tag_identifier.py

```diff
@@ -1,170 +1,172 @@
-    classifier to predict labels for the identifier tokens.
-
-    Args:
-        clf (Classifier): The trained classifier model.
-        data (pd.DataFrame): A DataFrame containing features for identifier tokens. The columns of the DataFrame should
-                             match the feature names used during training.
-
-    Returns:
-        np.array: An array of predicted labels for the identifier tokens.
-
-    Example:
-        clf = load_classifier('trained_classifier.pkl')
-        data = pd.DataFrame(...)  # Create a DataFrame with feature data.
-        predictions = annotate_identifier(clf, data)
-    """
-    
-    independent_variables_add = ['NORMALIZED_POSITION', 'LAST_LETTER', 'CONTEXT', 'MAXPOSITION', 'NLTK_POS', 'POSITION',
-                                 'VERB_SCORE', 'DET_SCORE', 'PREP_SCORE', 'CONJ_SCORE', 'PREPOSITION', 'DETERMINER',
-                                 'ENGLISHV_SCORE', 'ENGLISHN_SCORE', 'METHODN_SCORE', 'METHODV_SCORE', 'CODEPRE_SCORE',
-                                 'METHODPRE_SCORE', 'ENGLISHPRE_SCORE']
-    
-    df_features = pd.DataFrame(data, columns=independent_variables_add)
-    y_pred = clf.predict(df_features)
-    return y_pred
\ No newline at end of file
+import os, joblib
+import pandas as pd
+from feature_generator import *
+from flask import Flask
+from waitress import serve
+from spiral import ronin
+
+app = Flask(__name__)
+
+class ModelData:
+    def __init__(self, ModelTokens, ModelMethods, ModelGensimEnglish) -> None:
+        """
+        Initialize an instance of the ModelData class with word vector models.
+
+        Args:
+            ModelTokens: Word vectors model for tokens.
+            ModelMethods: Word vectors model for methods.
+            ModelGensimEnglish: Word vectors model for general English words.
+        """
+        self.ModelTokens = ModelTokens
+        self.ModelMethods = ModelMethods
+        self.ModelGensimEnglish = ModelGensimEnglish
+
+def initialize_model():
+    """
+    Initialize and load word vectors for the application.
+
+    This function initializes and loads word vectors using the 'createModel' function and stores them in the
+    'app.model_data' attribute for later use.
+
+    Returns:
+        None
+    """
+    print("Loading word vectors!!")
+    ModelTokens, ModelMethods, ModelGensimEnglish = createModel()
+    print("Word vectors loaded!!")
+
+    app.model_data = ModelData(ModelTokens, ModelMethods, ModelGensimEnglish)
+
+def start_server():
+    """
+    Initialize the model and start the server.
+
+    This function first initializes the model by calling the 'initialize_model' function. Then, it starts the server using
+    the Flask 'app.run' method, allowing incoming HTTP requests to be handled.
+
+    Returns:
+        None
+    """
+    initialize_model()
+    print("Starting server!!")
+    #app.run(host='0.0.0.0')
+    serve(app, host='127.0.0.1', port=5000)

```

Comment:
Should this be 127.0.0.1? Fine, if so. Just want confirmation.

Reply:
The arguments passed to the `serve()` function tells waitress what IP address and port it should use to look for requests. In this case, I had accidentally left the loopback address I was using to test as the argument. By setting this argument to `0.0.0.0`, waitress will look for requests on any interface with an IP address. In production, this is something that should probably be configurable. 

I've just committed some changes in my fork which allow for this information to be configured in the `serve.json` file, or through command line arguments: `python3 main.py -c host port protocol` where protocol can specify whether the tagger uses HTTP or HTTPS.